### ISSUE TITLE
Gaussian Process-based Bayesian Search estimator is different between gp_minimize and BayesSearchCV

### ISSUE THREAD

**Issue by pavelkomarov** [scikit-optimize/scikit-optimize#585]
Building on my last issue (#583), I wrote my own `base_minimize` that takes dictionary-defined spaces and parallelizes execution of the objective function. I am now writing custom functions on top to do things like Bayesian search, and tracing your code I notice that `gp_minimize` creates a normalized `Space` object and passes this to `base_minimize`, while in `BayesSearchCV` the `params_space` is passed to `_make_optimizer`, which passes a listified *and not normalized* version of that dictionary-defined space down to `Optimizer`'s constructor.

1. Is there a good reason to use the normalized space, and if so why is that not being done in BayesSearchCV?
2. Won't gp_minimize return results in the transformed, normalized space? Having to transform back to unnormalized space isn't so nice if I'm looking for hyperparameter values.

**Comment by pavelkomarov** [comment]
Line 179 of optimizer.py:

    if isinstance(self.base_estimator_, GaussianProcessRegressor):
        dimensions = normalize_dimensions(dimensions)

So it appears when the string 'gp' is passed as `base_estimator`, then when the estimator is "cooked" in `utils` it is determined to be and returned as a `GaussianProcessRegressor`, so the dimensions will get normalized either way.

This means that extra code to intentionally normalize the space in `gp_minimize` is unnecessary and out of date.

And the question remains: Where does the inverse transform happen?

**Comment by betatim** [comment]
> This means that extra code to intentionally normalize the space in gp_minimize is unnecessary and out of date.

PRs are welcome to help keep things moving forwards.

The surrogate model is only ever given points in the transformed space (sometimes the transform is the identity function). The objective function only ever gets given points from the original space. You can see part of the conversion happening in [`Optimizer.tell`](https://github.com/scikit-optimize/scikit-optimize/blob/6deff296237422c8b60a5dadb1336dc5e1684c93/skopt/optimizer/optimizer.py#L453) for example.

All optimisation methods (`*_minimize` and `BayesSearchCV`) eventually use `Optimizer`'s `ask` and `tell` methods so they should all be doing the same thing.

**Comment by betatim** [comment]
As to why transform dimensions when using a gaussian process you will have to follow up in the research literature. Empirically it works better, similar to normalising samples that are fitted by a neural network.

**Comment by pavelkomarov** [comment]
Right, I thought normalizing for GPRegression must be a good idea. Things tend to do better when features are all on the same scale.

Good to know the transform and inverse transform are actually being utilized deep in `Optimizer` and that the user should never have to worry about the transformed space.

Yes, I should remove the obsolete code and push it up. Do you have a set of tests for these things so I can quickly ensure I haven't broken anything?

**Comment by betatim** [comment]
I just tried it and if all you do is remove the call to `normalize_dimensions` in `gp_minimize` some tests start failing. So there is probably a reason for it being there.

**Comment by pavelkomarov** [comment]
Does anyone have any idea why a seemingly redundant normalization is there? Are the failing tests obsolete or something? I'll have to try it for myself.

**Comment by betatim** [comment]
One of the jobs for our tests is to ensure that things that used to work continue to work/don't change behaviour. So I think the way to find out is to look at each test that fails and analyse why. Modifying a test is one of the hardest bits because they encode a large part of our collective knowledge/experience and assumptions (so that it doesn't have to be in people's heads).

My hunch is that how the kernel for the GP regressor is chosen is not quite right/makes assumptions about what happens when.

It isn't on my list of priorities to find this, but it would be good to figure out what is going on.
